[highlighter] StackOverflowError when querying against SpatialRecursivePrefixTreeFieldType with highlighting

Solr throws ``java.lang.StackOverflowError`` when ``TaggedQueryHighlighter`` is enabled and query is done against ``SpatialRecursivePrefixTreeFieldType`` value with highlighting on any field. 

schema.xml:
```xml 
<?xml version="1.0" encoding="UTF-8" ?>
<schema name="core_name" version="1.4">
    <types>
        <!-- .... -->
        <fieldType name="location_rpt" class="solr.SpatialRecursivePrefixTreeFieldType" units="degrees" geo="true" />
        <!-- .... -->
    </types>
    <fields>
        <!-- .... -->
        <field name="location" type="location_rpt" indexed="true" stored="true"/>
        <!-- .... -->
    </fields>
</schema>
```

Field value is in "lat,long" format, i.e. ``56.9484,24.0981``.

Query:
```
q={!geofilt sfield="location" pt="56.9484,24.0981" d=100}&hl.fl=documentId
```

Response:
```
{
  "error":{
    "msg":"java.lang.StackOverflowError",
    "trace":"java.lang.RuntimeException: java.lang.StackOverflowError\n\tat org.apache.solr.servlet.SolrDispatchFilter.sendError(SolrDispatchFilter.java:790)\n\tat org.apache.solr.servlet.SolrDispatchFilter.doFilter(SolrDispatchFilter.java:439)\n\tat org.apache.solr.servlet.SolrDispatchFilter.doFilter(SolrDispatchFilter.java:207)\n\tat org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:235)\n\tat org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:206)\n\tat org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:233)\n\tat org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:191)\n\tat org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:127)\n\tat org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:102)\n\tat org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:109)\n\tat org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:293)\n\tat org.apache.coyote.http11.Http11Processor.process(Http11Processor.java:859)\n\tat org.apache.coyote.http11.Http11Protocol$Http11ConnectionHandler.process(Http11Protocol.java:602)\n\tat org.apache.tomcat.util.net.JIoEndpoint$Worker.run(JIoEndpoint.java:489)\n\tat java.lang.Thread.run(Thread.java:722)\nCaused by: java.lang.StackOverflowError\n\tat com.google.common.collect.UnmodifiableIterator.<init>(UnmodifiableIterator.java:32)\n\tat com.google.common.collect.UnmodifiableListIterator.<init>(UnmodifiableListIterator.java:34)\n\tat com.google.common.collect.AbstractIndexedListIterator.<init>(AbstractIndexedListIterator.java:68)\n\tat com.google.common.collect.Iterators$12.<init>(Iterators.java:1125)\n\tat com.google.common.collect.Iterators.forArray(Iterators.java:1125)\n\tat com.google.common.collect.RegularImmutableList.listIterator(RegularImmutableList.java:96)\n\tat com.google.common.collect.ImmutableList.listIterator(ImmutableList.java:334)\n\tat com.google.common.collect.ImmutableList.iterator(ImmutableList.java:330)\n\tat com.google.common.collect.ImmutableList.iterator(ImmutableList.java:62)\n\tat com.sematext.lucene.query.extractor.QueryExtractor.extractQuery(QueryExtractor.java:92)\n\tat com.sematext.lucene.query.extractor.ConstantScoreQueryExtractor.extract(ConstantScoreQueryExtractor.java:32)\n\tat com.sematext.lucene.query.extractor.ConstantScoreQueryExtractor.extract(ConstantScoreQueryExtractor.java:14)\n\tat com.sematext.lucene.query.extractor.QueryExtractor.extractQuery(QueryExtractor.java:94)\n\tat com.sematext.lucene.query.extractor.ConstantScoreQueryExtractor.extract(ConstantScoreQueryExtractor.java:32)\n\tat com.sematext.lucene.query.extractor.ConstantScoreQueryExtractor.extract(ConstantScoreQueryExtractor.java:14)\n\tat com.sematext.lucene.query.extractor.QueryExtractor.extractQuery(QueryExtractor.java:94)\n\tat com.sematext.lucene.query.extractor.ConstantScoreQueryExtractor.extract(ConstantScoreQueryExtractor.java:32)\n\tat com.sematext.lucene.query.extractor.ConstantScoreQueryExtractor.extract(ConstantScoreQueryExtractor.java:14)\n\tat com.sematext.lucene.query.extractor.QueryExtractor.extractQuery(QueryExtractor.java:94)\n\tat com.sematext.lucene.query.extractor.ConstantScoreQueryExtractor.extract(ConstantScoreQueryExtractor.java:32)\n\tat com.sematext.lucene.query.extractor.ConstantScoreQueryExtractor.extract(ConstantScoreQueryExtractor.java:14)\n\tat com.sematext.lucene.query.extractor.QueryExtractor.extractQuery(QueryExtractor.java:94)\n\tat com.sematext.lucene.query.extractor.ConstantScoreQueryExtractor.extract(ConstantScoreQueryExtractor.java:32)\n\tat com.sematext.lucene.query.extractor.ConstantScoreQueryExtractor.extract(ConstantScoreQueryExtractor.java:14)\n\tat com.sematext.lucene.query.extractor.QueryExtractor.extractQuery(QueryExtractor.java:94)\n\tat 
........
",
    "code":500}}
```

--- a/src/main/java/com/sematext/lucene/query/extractor/ConstantScoreQueryExtractor.java
+++ b/src/main/java/com/sematext/lucene/query/extractor/ConstantScoreQueryExtractor.java
@@ -26,10 +26,8 @@ public class ConstantScoreQueryExtractor extends QueryExtractor<ConstantScoreQue
           final List<Query> extractedQueries) throws UnsupportedOperationException {
     if (q.getQuery() != null) {
       extractQuery(q.getQuery(), extractors, extractedQueries);
-    }
-    else
-    {
-      extractQuery(q, extractors, extractedQueries);
+    } else {
+      extractedQueries.add(q);
     }
   }
 

--- a/src/test/java/com/sematext/solr/highlighter/TestTaggedQueryHighlighterIT.java
+++ b/src/test/java/com/sematext/solr/highlighter/TestTaggedQueryHighlighterIT.java
@@ -164,4 +164,17 @@ public class TestTaggedQueryHighlighterIT extends SolrTestCaseJ4 {
             "count(//lst[@name='highlighting']/*)=1",
             "//lst[@name='highlighting']/lst[@name='1']/arr[@name='test_tag1']/str='<em>test1</em>'");
   }
+
+  @Test
+  public void testConstantScoreQueryWithFilterPartOnly() {
+    final String[] doc1 = {"id", "1", "location", "56.9485,24.0980"};
+    assertU(adoc(doc1));
+    assertU(commit());
+
+    ModifiableSolrParams params = new ModifiableSolrParams();
+    params.add("q", "{!geofilt sfield=\"location\" pt=\"56.9484,24.0981\" d=100}");
+    params.add("hl", "true");
+    params.add("hl.fl", "location");
+    assertQ(req(params), "*[count(//doc)=1]", "count(//lst[@name='highlighting']/*)=1");
+  }
 }

--- a/src/test/resources/solr/collection1/conf/schema.xml
+++ b/src/test/resources/solr/collection1/conf/schema.xml
@@ -34,12 +34,14 @@
   <fieldType name="string" class="solr.StrField" sortMissingLast="true" omitNorms="true"/>
   <fieldType name="double" class="solr.TrieDoubleField" sortMissingLast="true" omitNorms="true"/>
   <fieldType name="int" class="solr.TrieIntField" sortMissingLast="true" omitNorms="true"/>
+  <fieldType name="location_rpt" class="solr.SpatialRecursivePrefixTreeFieldType" units="degrees" geo="true" />
 
   <field name="_version_" type="string"/>
   <field name="id" type="string" required="true"/>
   <field name="string_field" type="string"/>
   <field name="double_field" type="double"/>
   <field name="int_field" type="int"/>
+  <field name="location" type="location_rpt" />
 
   <!-- Field to use to determine and enforce document uniqueness.
       Unless this field is marked with required="false", it will be a required field

